File: java/Hash/longestConsecutive/longestConsecutive.java
import java.util.HashSet;
import java.util.Set;

public class longestConsecutive {

    public int longestConsecutiveQ(int[] nums) {
        Set<Integer> numSet = new HashSet<>();
        for (var num : nums) {
            numSet.add(num);
        }

        int MaxLen = 0;

        for (var num : numSet) {
            if (!numSet.contains(num - 1)) {
                int currentNum = num;
                int count = 1;

                while (numSet.contains(currentNum + 1)) {
                    currentNum++;
                    count++;
                }

                if (count > MaxLen) {
                    MaxLen = count;
                }
            }
        }
        return 0;
    }
}

File: golang/Hash/longestConsecutive/longestConsecutive.go
package longestconsecutive

func longestConsecutive(nums []int) int {
	numsSet := make(map[int]struct{})

	for _, num := range nums {
		numsSet[num] = struct{}{}
	}

	maxLen := 0

	for num := range numsSet {
		if _, ok := numsSet[num-1]; !ok {
			currentNum := num
			count := 1

			for {
				if _, isExis := numsSet[currentNum+1]; !isExis {
					break
				}
				currentNum++
				count++
			}

			maxLen = max(count, maxLen)
		}
	}

	return maxLen
}
